docs/contributing/guidelines-for-phing-targets.md

```diff
@@ -0,0 +1,89 @@
+# Guidelines for Phing Targets
+
+There are a few rules to help us maintain Phing targets in Shopsys Framework.
+
+## Naming conventions
+
+Names of targets should be in lower-case, using a dash (`-`) as a word separator.
+
+Names of properties should be also in lower-case, using a dot (`.`) to separate levels of hierarchy and a dash (`-`) as a word separator.
+
+When a target enforces some application environment, it should have the environment as a prefix (eg. `test-db-create` or `prod-warmup`).
+Targets without such prefix should use the current environment.
+
+## Dependent phing targets vs. phingcall
+
+There are two ways how to combine phing targets together: using `depends="..."` attribute or the `<phingcall .../>` task.
+
+You can set a dependency between target using the `depends` property like this:
+
+```xml
+<target name="my-target" depends="first-target,second-target">
+    <!-- implementation -->
+</target>
+```
+
+This will mean that before the execution of `my-target`, the targets `first-target` and `second-target` will be executed as well.
+
+This is also useful when you want to implement a target that runs a sequence of subtargets.
+Typically, targets starting with a `build-` prefix are such sequences.
+
+On the other hand you can use a `<phingcall .../>` task inside you implementation to run a specific target:
+
+```xml
+<target name="my-target">
+    <!-- implementation -->
+    <phingcall target="sub-target"/>
+    <!-- implementation -->
+</target>
+```
+
+This will execute the `sub-target` task during the execution of `my-target`.
+All dependencies of `my-target` will be called as well, even if they were already called during the execution.
+Because of this, `phingcall` is not suitable for implementation of sequences as the dependencies get called many times, making the build run longer.
+
+Still, it can be very useful when you need to control the exact time when it is called, or you want to run it with some properties overwritten.
+
+## Dependency on DIC
+
+Phing targets that use application dependency injection container (i.e. targets that execute Symfony commands) must be run after `composer install` is run.
+
+This means that inside targets that include a dependency on `composer-dev` or `composer-prod`, these DIC-dependent tasks must not be executed before Composer is run.
+
+## Extensibility
+
+When implementing a phing target inside the `shopsys/framework` package, please remember that the end users may have different needs.
+
+Try to avoid hardcoding values that can be replaced by a property.
+Such properties can be overwritten in a project repository without the need for overwriting the whole target.
+
+For example, all paths to executables should be used via `path.*.executable` property.
+
+Sometimes, it can be even useful to include a `<arg line="${my-target.flags}"/>` in an `exec` task with an empty `my-target.flags` property provided.
```

Comment:
what about like this?
```suggestion
Sometimes, it can be even useful to include `<arg line="${my-target.flags}"/>` in `exec` task with an empty `my-target.flags` property provided.  
```

Reply:
I've revised it a bit in 30d836a12, but I think that just removing the articles or adding a line-break don't help much.